src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/FluentWindow.cs

```diff
@@ -0,0 +1,231 @@
+
+    /// <summary>
+    /// This virtual method is called when <see cref="WindowCornerPreference"/> is changed.
+    /// </summary>
+    protected virtual void OnCornerPreferenceChanged(
+        WindowCornerPreference oldValue,
+        WindowCornerPreference newValue
+    )
+    {
+        if (InteropHelper.Handle == IntPtr.Zero)
+            return;
+
+        UnsafeNativeMethodsWindow.ApplyWindowCornerPreference(InteropHelper.Handle, newValue);
+    }
+
+    /// <summary>
+    /// Private <see cref="WindowBackdropType"/> property callback.
+    /// </summary>
+    private static void OnBackdropTypeChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+    {
+        if (d is not FluentWindow window)
+            return;
+
+        if (e.OldValue == e.NewValue)
+            return;
+
+        window.OnBackdropTypeChanged((WindowBackdropType)e.OldValue, (WindowBackdropType)e.NewValue);
+    }
+
+    /// <summary>
+    /// This virtual method is called when <see cref="WindowBackdropType"/> is changed.
+    /// </summary>
+    protected virtual void OnBackdropTypeChanged(WindowBackdropType oldValue, WindowBackdropType newValue)
+    {
+        if (Appearance.ApplicationThemeManager.GetAppTheme() == Appearance.ApplicationTheme.HighContrast)
+        {
+            newValue = WindowBackdropType.None;
+        }
+
+        if (InteropHelper.Handle == IntPtr.Zero)
+            return;
+
+        if (newValue == WindowBackdropType.None)
+        {
+            WindowBackdrop.RemoveBackdrop(this);
+            return;
+        }
+
+        if (!ExtendsContentIntoTitleBar)
+            throw new InvalidOperationException(
+                $"Cannot apply backdrop effect if {nameof(ExtendsContentIntoTitleBar)} is false."
+            );
+
+        if (WindowBackdrop.IsSupported(newValue) && WindowBackdrop.RemoveBackground(this))
+            WindowBackdrop.ApplyBackdrop(this, newValue);
+    }
+
+    /// <summary>
+    /// Private <see cref="ExtendsContentIntoTitleBar"/> property callback.
+    /// </summary>
+    private static void OnExtendsContentIntoTitleBarChanged(
+        DependencyObject d,
+        DependencyPropertyChangedEventArgs e
+    )
+    {
+        if (d is not FluentWindow window)
+            return;
+
+        if (e.OldValue == e.NewValue)
+            return;
+
+        window.OnExtendsContentIntoTitleBarChanged((bool)e.OldValue, (bool)e.NewValue);
+    }
+
+    /// <summary>
+    /// This virtual method is called when <see cref="ExtendsContentIntoTitleBar"/> is changed.
+    /// </summary>
+    protected virtual void OnExtendsContentIntoTitleBarChanged(bool oldValue, bool newValue)
+    {
+        WindowStyle = WindowStyle.SingleBorderWindow;
+        //AllowsTransparency = true;
+
+        WindowChrome.SetWindowChrome(
+            this,
+            new WindowChrome
+            {
+                CaptionHeight = 0,
+                CornerRadius = default,
+                GlassFrameThickness = new Thickness(-1),
```

Comment:
After we set the GlassFrameThickness to -1, we will find the CloseButton do not fitted  the edge of the window.

![](https://github.com/dotnet/wpf/assets/16054566/142c049f-2519-495c-a2c8-13f1f8c55cb2)

```xml
<WindowChrome.WindowChrome>
    <WindowChrome GlassFrameThickness="-1" />
</WindowChrome.WindowChrome>
```